src/com/owncloud/android/ui/activity/ShareActivity.java

```diff
@@ -201,12 +197,25 @@ public void onBackPressed() {
     @Override
     public void onRemoteOperationFinish(RemoteOperation operation, RemoteOperationResult result) {
         super.onRemoteOperationFinish(operation, result);
-        if (operation instanceof UnshareOperation) {
-            refreshUsersInLists();
-        } else if(operation instanceof CreateShareWithShareeOperation){
-            refreshUsersInLists();
-            // Clean action
-            getIntent().setAction(null);
+        if (operation instanceof UnshareOperation ||
+                operation instanceof CreateShareWithShareeOperation) {
+
+            if (result.isSuccess()) {
```

Comment:
I'll change ```onGetDataShareWithFinish```. I prefer the app crashes in this case, helps to detect a development error sooner.